src/TestData/Factories/DonationFormFactory.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace GiveTestData\TestData\Factories;
+
+use GiveTestData\TestData\Framework\Factory;
+
+/**
+ * Class DonationFormFactory
+ * @package GiveTestData\TestData\Factories
+ */
+class DonationFormFactory extends Factory {
+
+	/**
+	 * @var bool
+	 */
+	private $donationGoal;
+
+	/**
+	 * @var bool
+	 */
+	private $termsAndConditions;
+
+	/**
+	 * @var string
+	 */
+	private $template;
+
+	/**
+	 * @var string[]
+	 */
+	private $templates = [ 'sequoia', 'legacy' ];
+
+	/**
+	 * @param string $template
+	 *
+	 * @return bool
+	 */
+	public function checkFormTemplate( $template ) {
+		if ( 'random' === $template ) {
+			return true;
+		}
+
+		return in_array( $template, $this->templates );
+	}
+
+	/**
+	 * @param string $template
+	 */
+	public function setFormTemplate( $template ) {
+		$this->template = $template;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getFormTemplate() {
+		if ( 'random' === $this->template ) {
+			return $this->randomDonationTemplate();
+		}
+
+		return $this->template;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function randomDonationTemplate() {
+		return $this->faker->randomElement( $this->templates );
+	}
+
+	/**
+	 * @param bool $generate
+	 */
+	public function setDonationFormGoal( $generate ) {
+		$this->donationGoal = ( bool ) $generate;
+	}
+
+	/**
+	 * @return false|string
+	 */
+	public function getDonationGoal() {
+		if ( is_null( $this->donationGoal ) || ! $this->donationGoal ) {
+			return false;
+		}
+
+		return $this->randomGoal();
+	}
+
+	/**
+	 * @param bool $generate
+	 */
+	public function setTermsAndConditions( $generate ) {
+		$this->termsAndConditions = ( bool ) $generate;
+	}
+
+
+	/**
+	 * @return array
+	 */
+	public function getTermsAndConditions() {
+		if ( is_null( $this->termsAndConditions ) || ! $this->termsAndConditions ) {
+			return [];
+		}
+
+		return [
```

Comment:
This could be extracted to a new provider.

Reply:
Yes, definitely.